execution/block-partitioner/src/v2/mod.rs

```diff
@@ -0,0 +1,719 @@
+                                let anchor_shard_id =
+                                    get_anchor_shard_id(storage_location, num_executor_shards);
+                                RwLock::new(ConflictingTxnTracker::new(
+                                    storage_location.clone(),
+                                    anchor_shard_id,
+                                ))
+                            })
+                            .write()
+                            .unwrap()
+                            .add_candidate(txn_idx, is_write);
+                    }
+                });
+        });
+        let duration_1 = timer_1.stop_and_record();
+        info!("preprocess__main={duration_1}");
+        let duration = timer.stop_and_record();
+        info!("preprocess={duration}");
+
+        let timer = MISC_TIMERS_SECONDS
+            .with_label_values(&["pre_partition_uniform"])
+            .start_timer();
+        let remaining_txn_idxs = uniform_partition(num_txns, num_executor_shards);
+        let mut start_txns: Vec<OriginalTxnIdx> = vec![0; num_executor_shards];
+        for shard_id in 1..num_executor_shards {
+            start_txns[shard_id] =
+                start_txns[shard_id - 1] + remaining_txn_idxs[shard_id - 1].len();
+        }
+        let duration = timer.stop_and_record();
+        info!("pre_partition_uniform={duration}");
+
+        let timer = MISC_TIMERS_SECONDS
+            .with_label_values(&["multi_rounds"])
+            .start_timer();
+        let (finalized_txn_matrix, start_index_matrix, new_idxs) = self.multi_rounds(
+            num_txns,
+            num_executor_shards,
+            &rsets,
+            &wsets,
+            &senders,
+            &trackers,
+            &start_txns,
+            remaining_txn_idxs,
+        );
+        let duration = timer.stop_and_record();
+        info!("multi_rounds={duration}");
+
+        let timer = MISC_TIMERS_SECONDS
+            .with_label_values(&["add_edges"])
+            .start_timer();
+        let ret = self.add_edges(
+            txns,
+            &rsets,
+            &wsets,
+            &finalized_txn_matrix,
+            &start_index_matrix,
+            &new_idxs,
+            &trackers,
+        );
+        let duration = timer.stop_and_record();
+        info!("add_edges={duration}");
+        let timer = MISC_TIMERS_SECONDS
+            .with_label_values(&["drop"])
+            .start_timer();
+        self.thread_pool.spawn(move || {
+            drop(sender_idx_table);
+            drop(senders);
+            drop(wsets);
+            drop(rsets);
+            drop(trackers);
+            drop(key_idx_table);
+            drop(start_txns);
+            drop(finalized_txn_matrix);
+            drop(start_index_matrix);
+            drop(new_idxs);
+        });
+        let duration = timer.stop_and_record();
+        info!("drop={duration}");
+        ret
+    }
+}
+
+#[test]
+fn test_partitioner_v2_correctness() {
```

Comment:
Let's move all tests into a test module. Also, can you move all V1 tests to a common module and ensure the relevant tests work with both V1 and V2?

Reply:
i'd like to keep struct and tests in the same file. Now all the files are much smaller, that should not hurt readability any more?